src/approaches/interactive_learning_approach.py

```diff
@@ -117,7 +121,13 @@ def _relearn_predicates_and_nsrts(
                 (self._predicates_to_learn - {pred}) | {new_pred}
 
         # Learn NSRTs via superclass
-        self._learn_nsrts(dataset.trajectories, online_learning_cycle)
+        self._learn_nsrts(self._dataset.trajectories, online_learning_cycle)
+
+        # Save the things we need other than the NSRTs, which were already
+        # saved in the above call to self._learn_nsrts()
+        save_path = utils.get_approach_save_path_str()
+        with open(f"{save_path}_{online_learning_cycle}.DATA", "wb") as f:
+            pkl.dump((self._dataset, self._predicates_to_learn), f)
```

Comment:
instead of a tuple, can we make this a dict with keys `dataset, predicates_to_learn`?